x-pack/plugins/security_solution/public/detections/components/rules/duration_input/index.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFieldNumber,
+  EuiFormRow,
+  EuiSelect,
+  EuiFormControlLayout,
+  transparentize,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+
+import type { FieldHook } from '../../../../shared_imports';
+import { getFieldValidityAndErrorMessage } from '../../../../shared_imports';
+import * as I18n from './translations';
+
+interface DurationInputProps {
+  durationValueField: FieldHook<number | undefined>;
+  durationUnitField: FieldHook<string>;
+  minimumValue?: number;
+  isDisabled: boolean;
+  durationUnitOptions?: Array<{ value: 's' | 'm' | 'h' | 'd'; text: string }>;
+}
+
+const getNumberFromUserInput = (input: string, minimumValue = 0): number | undefined => {
+  const number = parseInt(input, 10);
+  if (Number.isNaN(number)) {
+    return minimumValue;
+  } else {
+    return Math.max(minimumValue, Math.min(number, Number.MAX_SAFE_INTEGER));
+  }
+};
+
+const StyledEuiFormRow = styled(EuiFormRow)`
+  max-width: none;
+
+  .euiFormControlLayout {
+    max-width: 235px;
+    width: auto;
+  }
+
+  .euiFormControlLayout__childrenWrapper > *:first-child {
+    box-shadow: none;
+    height: 38px;
+    width: 100%;
+  }
+
+  .euiFormControlLayout__childrenWrapper > select {
+    background-color: ${({ theme }) => transparentize(theme.eui.euiColorPrimary, 0.1)};
+    color: ${({ theme }) => theme.eui.euiColorPrimary};
+  }
+
+  .euiFormControlLayout--group .euiFormControlLayout {
+    min-width: 100px;
+  }
+
+  .euiFormControlLayoutIcons {
+    color: ${({ theme }) => theme.eui.euiColorPrimary};
+  }
+
+  .euiFormControlLayout:not(:first-child) {
+    border-left: 1px solid ${({ theme }) => theme.eui.euiColorLightShade};
+  }
+`;
+
+const MyEuiSelect = styled(EuiSelect)`
+  width: auto;
+`;
+
+// This component is similar to the ScheduleItem component, but instead of combining the value
+// and unit into a single string it keeps them separate. This makes the component simpler and
+// allows for easier validation of values and units in APIs as well.
```

Comment:
Thanks for this comment -- was going to ask if we could re-use the `ScheduleItem` component but makes sense wanting to keep them separate for simplicities sake. Would be nice to refactor `ScheduleItem` and add support for keeping the value/unit separate to keep it more reusable, but we can go down that road later when refactoring these forms.

Reply:
Yeah it would be nice to remove the combined value+unit strings like the rule schedule IMO. It makes the validation easier with less edge cases to worry about.